### PR TITLE
[Access] 4506 - Make CCF encoded events optional via request parameter

### PR DESCRIPTION
**Pull request by Guitarheroua**
https://github.com/onflow/flow-go/issues/4506

### Context

This pull request introduces a new request parameter for a list of endpoints from the [AccessAPI](https://github.com/onflow/flow-go/blob/master/access/api.go) and the [ExecutionDataAPI](https://github.com/onflow/flow-go/blob/dafd8fd29c476cacc497f79940412ba377f723fe/engine/access/state_stream/backend.go#L73), which specifies the encoding for events.

As part of this PR:

* The protobuf structures have been extended (see https://github.com/onflow/flow/pull/1391)
* main flow-go PR (see https://github.com/onflow/flow-go/pull/4809)

### CODE REVIEW

File: adapters/access.go [hunk 1]
```diff
@@ -197,11 +199,13 @@ func (a *AccessAdapter) GetTransactionResult(
...
-	result.Events, err = ConvertCCFEventsToJsonEvents(result.Events)
-	if err != nil {
-		return nil, convertError(err)
+	// Convert CCF events to JSON events, else return CCF encoded version
+	if requiredEventEncodingVersion == entities.EventEncodingVersion_JSON_CDC_V0 {
+		result.Events, err = ConvertCCFEventsToJsonEvents(result.Events)
+		if err != nil {
+			return nil, convertError(err)
+		}
 	}
```
Comment (turbolent) [inline adapters/access.go:208]: The old and new code seem repeated in multiple places. Maybe make this a function to avoid the repeated code

**Review by turbolent — COMMENTED** [review]
👍

**Comment by peterargue** [comment]
@Guitarheroua can you pull in master and resolve these conflicts?

**Review by peterargue — APPROVED** [review]
looks good to me

**Review by turbolent — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -190,3 +190,6 @@ require (
 	modernc.org/sqlite v1.21.1 // indirect
 	nhooyr.io/websocket v1.8.7 // indirect
 )
+
+//TODO: Remove when both version will be merged
+replace github.com/onflow/flow-go v0.32.3 => github.com/Guitarheroua/flow-go v0.0.0-20231024184136-247768e776f7
```
Comment (tarakby) [inline go.mod:195]: @peterargue do you know if we can remove this statement? does flow-go has the required changes?

Comment (peterargue) [reply go.mod:195]: yes, though we haven't tagged a release including this yet. should be tagged later this afternoon

**Pull request merged**